sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client_helpers.py

```diff
@@ -0,0 +1,1225 @@
+
+    encoding = kwargs.pop('encoding', 'UTF-8')
+    if isinstance(data, str):
+        data = data.encode(encoding)
+    if length is None:
+        length = get_length(data)
+    if isinstance(data, bytes):
+        data = data[:length]
+
+    if isinstance(data, bytes):
+        stream = BytesIO(data)
+    elif hasattr(data, 'read'):
+        stream = data
+    elif hasattr(data, '__iter__') and not isinstance(data, (list, tuple, set, dict)):
+        stream = IterStreamer(data, encoding=encoding)
+    elif hasattr(data, '__aiter__'):
+        stream = AsyncIterStreamer(data, encoding=encoding)
+    else:
+        raise TypeError(f"Unsupported data type: {type(data)}")
+
+    validate_content = kwargs.pop('validate_content', False)
+    content_settings = kwargs.pop('content_settings', None)
+    overwrite = kwargs.pop('overwrite', False)
+    max_concurrency = kwargs.pop('max_concurrency', 1)
+    cpk = kwargs.pop('cpk', None)
+    cpk_info = None
+    if cpk:
+        cpk_info = CpkInfo(encryption_key=cpk.key_value, encryption_key_sha256=cpk.key_hash,
+                            encryption_algorithm=cpk.algorithm)
+    kwargs['cpk_info'] = cpk_info
+
+    headers = kwargs.pop('headers', {})
+    headers.update(add_metadata_headers(metadata))
+    kwargs['lease_access_conditions'] = get_access_conditions(kwargs.pop('lease', None))
+    kwargs['modified_access_conditions'] = get_modify_conditions(kwargs)
+    kwargs['cpk_scope_info'] = get_cpk_scope_info(kwargs)
+    if content_settings:
+        kwargs['blob_headers'] = BlobHTTPHeaders(
+            blob_cache_control=content_settings.cache_control,
+            blob_content_type=content_settings.content_type,
+            blob_content_md5=content_settings.content_md5,
+            blob_content_encoding=content_settings.content_encoding,
+            blob_content_language=content_settings.content_language,
+            blob_content_disposition=content_settings.content_disposition
+        )
+    kwargs['blob_tags_string'] = serialize_blob_tags_header(kwargs.pop('tags', None))
+    kwargs['stream'] = stream
+    kwargs['length'] = length
+    kwargs['overwrite'] = overwrite
+    kwargs['headers'] = headers
+    kwargs['validate_content'] = validate_content
+    kwargs['blob_settings'] = config
+    kwargs['max_concurrency'] = max_concurrency
+    kwargs['encryption_options'] = encryption_options
+    # Add feature flag to user agent for encryption
+    if encryption_options['key']:
+        modify_user_agent_for_encryption(
+            config.user_agent_policy.user_agent,
+            sdk_moniker,
+            encryption_options['version'],
+            kwargs)
+
+    if blob_type == BlobType.BlockBlob:
+        kwargs['client'] = client.block_blob
+        kwargs['data'] = data
+    elif blob_type == BlobType.PageBlob:
+        if encryption_options['version'] == '2.0' and (encryption_options['required'] or encryption_options['key'] is not None): # pylint: disable=line-too-long
```

Comment:
Split the line instead of ignoring it being too long.

```py
if (encryption_options['version'] == '2.0' and
   (encryption_options['required'] or encryption_options['key'] is not None)):
```